wal/src/rocks_impl/manager.rs

```diff
@@ -566,7 +621,34 @@ impl Builder {
             None
         };
 
-        let db = DB::open(rocksdb_config, &self.wal_path)
+        let mut cf_opts = ColumnFamilyOptions::new();
+        if let Some(v) = self.write_buffer_size {
+            cf_opts.set_write_buffer_size(v);
+        }
+        if let Some(v) = self.max_write_buffer_number {
+            cf_opts.set_max_write_buffer_number(v);
+        }
+        if let Some(v) = self.level_zero_file_num_compaction_trigger {
+            cf_opts.set_level_zero_file_num_compaction_trigger(v);
+        }
+        if let Some(v) = self.level_zero_slowdown_writes_trigger {
+            cf_opts.set_level_zero_slowdown_writes_trigger(v);
+        }
+        if let Some(v) = self.level_zero_stop_writes_trigger {
+            cf_opts.set_level_zero_stop_writes_trigger(v);
+        }
+
+        // FIFO compaction strategy let rocksdb looks like a message queue.
+        if let Some(v) = self.fifo_compaction_max_table_files_size {
+            if v > 0 {
+                let mut fifo_opts = FifoCompactionOptions::new();
+                fifo_opts.set_max_table_files_size(v);
+                cf_opts.set_fifo_compaction_options(fifo_opts);
+                cf_opts.set_compaction_style(DBCompactionStyle::Fifo);
+            }
+        }
+
+        let db = DB::open_cf(rocksdb_config, &self.wal_path, vec![("default", cf_opts)])
```

Comment:
Define a const or use the const in rocksdb?

Reply:
fixed